llrt_core/src/vm.rs

```diff
@@ -197,6 +199,20 @@ where
     T: Loader + 'static,
 {
     fn load<'js>(&mut self, ctx: &Ctx<'js>, name: &str) -> Result<Module<'js, Declared>> {
+        // Workaround for some characteristics of node_modules
+        // 1. If only the extension is not specified, the extension is completed.
+        // 2. If only a directory is specified, it is assumed to load index.js under the directory.
+        let name = if name.contains("node_modules") && Path::new(name).is_dir() {
+            [".js", ".mjs", ".cjs"]
+                .iter()
+                .map(|ext| [name, ext].concat())
+                .find(|path| Path::new(path).exists())
+                .unwrap_or_else(|| [name, "/index.js"].concat())
+        } else {
+            name.to_string()
+        };
+        let name = name.as_str();
```

Comment:
I also wanted to implement the workaround logic for node_modules on the Resolver side, rather than adding it to these places. However, I had no choice but to add it because I saw a behavior inside the node_modules package where only the loader is executed without going through Resolver when Javascript imports are run further from there.

I may have implemented this incorrectly, so I'll do some more testing.

Reply:
The reason was that the original BinaryResolver implementation returned the value as resolved when normalized_path existed as a directory.

By clearly determining when normalized_path exists as a file and when it exists as a directory, we were able to resolve the path with CostomResolver and make the Loader clean again.